appcues/src/test/java/com/appcues/action/ActionProcessorTest.kt

```diff
@@ -72,42 +84,46 @@ class ActionProcessorTest : KoinTest {
         // WHEN
         actionProcessor.process(
             listOf(action1, stepAction1, action2, stepAction2, action3, action4, stepAction3, action5),
-            StepInteraction.InteractionType.BUTTON_TAPPED,
+            BUTTON_TAPPED,
             "Button 1"
         )
 
         // THEN
-        assertThat(actionQueue.listQueue).hasSize(9)
-        assertThat(actionQueue.listQueue.first()).isInstanceOf(StepInteractionAction::class.java)
-        with(actionQueue.listQueue.first() as StepInteractionAction) {
-            assertThat(interaction.properties["destination"]).isEqualTo("c")
-            assertThat(interaction.properties["category"]).isEqualTo("test")
-            assertThat(interaction.properties["text"]).isEqualTo("Button 1")
+        val properties = slot<Map<String, Any>>()
+        verify { analyticsTracker.track("appcues:v2:step_interaction", capture(properties), false, true) }
+        with(properties.captured["interactionData"] as Map<*, *>) {
+            assertThat(this["destination"]).isEqualTo("c")
+            assertThat(this["text"]).isEqualTo("Button 1")
+            assertThat(this["category"]).isEqualTo("test")
```

Comment:
i think you can avoid the `capture` and do
```
verify { analyticsTracker.track("appcues:v2:step_interaction", mapOf("destination" to "c", "text" to "Button 1", "category" to "test"), false, true) }
```
but not 100% sure, and this is fine too!

Reply:
we cant because the properties map is not just a map of the InteractionData, but all the other attributes related to this track event.